packages/account/src/providers/transaction-request/transaction-request.test.ts

```diff
@@ -248,4 +248,55 @@ describe('transactionRequestify', () => {
     expect(txRequest.outputs).toEqual(txRequestLike.outputs);
     expect(txRequest.witnesses).toEqual(txRequestLike.witnesses);
   });
+
+  it('should keep data from input in transaction request created [upgrade]', () => {
+    const txRequestLike: TransactionRequestLike = {
+      type: TransactionType.Upgrade,
+      inputs: [],
+      outputs: [],
+      bytecodeWitnessIndex: 0,
+      upgradePurpose: {
+        type: UpgradePurposeTypeEnum.ConsensusParameters,
+        checksum: ZeroBytes32,
+      },
+    };
+
+    const txRequest = transactionRequestify(txRequestLike);
+
+    if (
+      txRequest.type === TransactionType.Upgrade &&
+      txRequest.upgradePurpose.type === UpgradePurposeTypeEnum.ConsensusParameters &&
+      txRequestLike.upgradePurpose?.type === UpgradePurposeTypeEnum.ConsensusParameters
+    ) {
+      expect(txRequest.upgradePurpose).toEqual(txRequestLike.upgradePurpose);
+      expect(txRequest.bytecodeWitnessIndex).toEqual(0);
+      expect(txRequest.upgradePurpose.checksum).toEqual(txRequestLike.upgradePurpose?.checksum);
+    }
+
+    expect(txRequest.type).toEqual(txRequestLike.type);
+  });
+
+  it('should keep data from input in transaction request created [upload]', () => {
+    const txRequestLike: TransactionRequestLike = {
+      type: TransactionType.Upload,
+      inputs: [],
+      outputs: [],
+      witnessIndex: 0,
+      subsection: {
+        root: ZeroBytes32,
+        subsectionIndex: 0,
+        subsectionsNumber: 1,
+        proofSet: [],
+      },
+    };
+
+    const txRequest = transactionRequestify(txRequestLike);
+
+    if (txRequest.type === TransactionType.Upload) {
+      expect(txRequest.subsection).toEqual(txRequestLike.subsection);
+      expect(txRequest.witnessIndex).toEqual(txRequestLike.witnessIndex);
+    }
```

Comment:
As above should we be throwing outside of the condition otherwise the assertions will be missed. You could cast on line `293` - `const txRequest = transactionRequestify(txRequestLike) as UploadTransactionRequest` if you're trying to avoid ts errors.